media-libs/openjpeg/openjpeg-2.5.0.ebuild

```diff
@@ -0,0 +1,141 @@
+	sys-libs/zlib:="
+DEPEND="${RDEPEND}"
+BDEPEND="
+	doc? ( app-doc/doxygen )"
+
+DOCS=( AUTHORS.md CHANGELOG.md NEWS.md README.md THANKS.md )
+
+PATCHES=(
+	"${FILESDIR}/${PN}-2.5.0-gnuinstalldirs.patch" # bug 667150
+)
+
+src_prepare() {
+	if use test; then
+		mv "${WORKDIR}"/openjpeg-data-${MY_TESTDATA_COMMIT} "${WORKDIR}"/data ||
+			die "Failed to rename test data"
+	fi
+
+	cmake_src_prepare
+}
+
+multilib_src_configure() {
+	local mycmakeargs=(
+		-DBUILD_PKGCONFIG_FILES=ON # always build pkgconfig files, bug #539834
+		-DBUILD_TESTING="$(multilib_native_usex test)"
+		-DBUILD_DOC=$(multilib_native_usex doc ON OFF)
+		-DBUILD_CODEC=$(multilib_is_native_abi && echo ON || echo OFF)
+		-DBUILD_STATIC_LIBS=$(usex static-libs)
+	)
+
+	# Cheat a little bit and force disabling fixed point magic
+	# The test suite is extremely fragile to small changes
+	# bug 715130, bug 715422
+	# https://github.com/uclouvain/openjpeg/issues/1017
+	multilib_is_native_abi && use test && append-cflags "-ffp-contract=off"
+
+	cmake_src_configure
+}
+
+multilib_src_test() {
+	if ! multilib_is_native_abi ; then
+		elog "Cannot run tests for non-multilib abi."
+		return 0
+	fi
+
+	local myctestargs=
+
+	pushd "${BUILD_DIR}" > /dev/null || die
+	if [[ -e CTestTestfile.cmake ]] || die "Test suite not available! Check source!"
+
+	if [[ -n ${TEST_VERBOSE} ]]; then
+		myctestargs+=( --extra-verbose --output-on-failure )
+	fi
+
+	echo ctest "${myctestargs[@]}" "$@"
+	if ctest "${myctestargs[@]}" "$@" ; then
+		einfo "Tests succeeded."
+		popd > /dev/null || die
+		return 0
+	fi
+
+	local FAILEDTEST_LOG="${BUILD_DIR}/Testing/Temporary/LastTestsFailed.log"
+
+	if [[ ! -f "${FAILEDTEST_LOG}" ]]; then
+		# Should never happen
+		die "Cannot analyze test failures: LastTestsFailed.log is missing!"
+	fi
+
+	echo ""
+	einfo "Note: Upstream is maintaining a list of known test failures."
+	einfo "We will now compare our test results against this list and sort out any known failure."
+
+	local KNOWN_FAILURES_LIST="${T}/known_failures_compiled.txt"
+	cat "${S}/tools/travis-ci/knownfailures-all.txt" > "${KNOWN_FAILURES_LIST}" || die
+
+	local ARCH_SPECIFIC_FAILURES=
+	if use amd64 ; then
+		ARCH_SPECIFIC_FAILURES="$(find "${S}/tools/travis-ci/" -name 'knownfailures-*x86_64*.txt' -print0 | sort -z | tail -z -n 1 | tr -d '\0')"
+	elif use x86 || use arm || use arm64; then
+		ARCH_SPECIFIC_FAILURES="$(find "${S}/tools/travis-ci/" -name 'knownfailures-*i386*.txt' -print0 | sort -z | tail -z -n 1 | tr -d '\0')"
+	fi
+
+	if [[ -f "${ARCH_SPECIFIC_FAILURES}" ]]; then
+		einfo "Adding architecture specific failures (${ARCH_SPECIFIC_FAILURES}) to known failures list ..."
+		cat "${ARCH_SPECIFIC_FAILURES}" >> "${KNOWN_FAILURES_LIST}" || die
+	fi
+
+	# Logic copied from $S/tools/travis-ci/run.sh
+	local FAILEDTEST=
+	local FAILURES_LOG="${BUILD_DIR}/Testing/Temporary/failures.txt"
+	local HAS_UNKNOWN_TEST_FAILURES=0
+
+	echo ""
+
+	awk -F: '{ print $2 }' "${FAILEDTEST_LOG}" > "${FAILURES_LOG}"
```

Comment:
Missing die.

Reply:
Changed it up so the output is piped straight into "while read". There's no intermediate file anymore.